apps/heft/src/cli/HeftCommandLineUtilities.ts

```diff
@@ -6,15 +6,21 @@ import {
   CommandLineFlagParameter,
   CommandLineStringParameter,
   CommandLineIntegerParameter,
-  CommandLineStringListParameter
+  CommandLineStringListParameter,
+  CommandLineParameter
 } from '@rushstack/ts-command-line';
 import { Terminal } from '@rushstack/node-core-library';
 
 /**
  * @beta
  * The base set of utility values provided in every object returned when registering a parameter.
  */
-export interface IHeftBaseParameter {
+export interface IHeftBaseParameter<TValue> {
```

Comment:
Should we include a `IHeftBaseParameter.definition` that lets you see information that was used to register it?  For example:

```ts
    const customStringParameter: IHeftStringParameter = heftSession.commandLine.registerStringParameter({
      associatedActionNames: ['build', 'test', 'start'],
      parameterLongName: '--custom-string-parameter',
      description: 'Test running a custom string parameter',
      argumentName: 'TEXT',
      required: true
    });
```

...if all I have is the `customStringParameter` object, it seems odd that I can't retrieve `--custom-string-parameter` or `required` from that object.

Reply:
True. I could see `required` and `associatedActionNames` being useful as a plugin developer.